packages/identity-secp256k1/src/pem.ts

```diff
@@ -0,0 +1,42 @@
+import { concat, uint8ToBuf } from '@dfinity/agent';
+
+const HEADER = `-----BEGIN`;
+const FOOTER = `-----END`;
+const ECHEADER = `-----BEGIN EC PRIVATE KEY`;
+
+/**
+ * Parse a PEM-encoded key into an ArrayBuffer
+ * @param pem - the PEM-encoded key
+ * @returns secret key as an ArrayBuffer
+ */
+export function pemToSecretKey(pem: string): ArrayBuffer {
+  const lines = pem.trim().split('\n');
+  const header = lines[0].trim();
+  const footer = lines[lines.length - 1].trim();
+  if (lines.length < 3) {
+    throw new Error('Invalid PEM format');
+  }
+  if (!header.startsWith(HEADER)) {
+    throw new Error('Invalid PEM header');
+  }
+  if (!footer.startsWith(FOOTER)) {
+    throw new Error('Invalid PEM footer');
+  }
```

Comment:
This will accept something like

```
----BEGIN EC PRIVATE KEY-----
// appropriate base64 encoding
-----END SURPRISE-----
```

or really even

```
----BEGIN EC PRIVATE KEY-----
// appropriate base64 encoding
-----END 
```

Generally you see functions like this operate in one of two ways:

* Takes the input string, plus the expected PEM label (eg "EC PRIVATE KEY"), and return the decoded bits. If the label doesn't match (checking both header and footer), return an error. This is useful for cases where you know what the PEM label is supposed to be.
* Take the input string, and return both the decoded bits plus the sniffed label (verifying it is used for both header + footer). This is useful where you might have to accept arbitrary labels, say for a pretty printer.

Often if the first style is used, it is defined in terms of the second style, eg (apologies for my psuedo-JS):

```
// TS has tuples right???
const (bits, label) = pem_decode(pemStr);
if label != expectedLabel {
  throw new Error(...);
}
return bits;
```

IIUC you only need the first style here.